workbench-core/example/infrastructure/integration-tests/tests/datasets/accessPermissions.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { dataSetPrefix } from '@aws/workbench-core-example-app/lib/configs/constants';
+import { CreateUser } from '@aws/workbench-core-user-management';
+import { v4 as uuidv4 } from 'uuid';
+import ClientSession from '../../support/clientSession';
+import Dataset from '../../support/resources/datasets/dataset';
+import Setup from '../../support/setup';
+import HttpError from '../../support/utils/HttpError';
+
+describe('DataSets access permissions integration tests', () => {
+  const setup: Setup = new Setup();
+  const mockBadValue: string = 'fake-data';
+  let adminSession: ClientSession;
+  let user: CreateUser;
+
+  beforeEach(() => {
+    expect.hasAssertions();
+  });
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+    user = {
+      firstName: 'Test',
+      lastName: 'User',
+      email: `success+create-user-${uuidv4()}@simulator.amazonses.com`
+    };
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('addDataSetAccessPermissions', () => {
+    it('Adds a read-only access permission.', async () => {
+      let response = await adminSession.resources.datasets.create({}, true);
+      const dataSetId: string = response.data.id;
+      response = await adminSession.resources.groups.create({}, true);
```

Comment:
we should keep these immutable for safety

Reply:
you mean the responses? In test code? I'll take a look.